src/libraries/System.IO.Hashing/src/System/IO/Hashing/XxHash128.cs

```diff
@@ -0,0 +1,460 @@
+            }
+        }
+
+        /// <summary>Resets the hash computation to the initial state.</summary>
+        public override void Reset()
+        {
+            XxHashShared.Reset(ref _state);
+        }
+
+        /// <summary>Appends the contents of <paramref name="source"/> to the data already processed for the current hash computation.</summary>
+        /// <param name="source">The data to process.</param>
+        public override void Append(ReadOnlySpan<byte> source)
+        {
+            XxHashShared.Append(ref _state, source);
+        }
+
+        /// <summary>Writes the computed 128-bit hash value to <paramref name="destination"/> without modifying accumulated state.</summary>
+        /// <param name="destination">The buffer that receives the computed hash value.</param>
+        protected override void GetCurrentHashCore(Span<byte> destination)
+        {
+            Hash128 current = GetCurrentHashAsHash128();
+            WriteBigEndian128(current, destination);
+        }
+
+        private Hash128 GetCurrentHashAsHash128()
+        {
+            Hash128 current;
+
+            if (_state.TotalLength > MidSizeMaxBytes)
+            {
+                // Digest on a local copy to ensure the accumulators remain unaltered.
+                ulong* accumulators = stackalloc ulong[AccumulatorCount];
+                CopyAccumulators(ref _state, accumulators);
+
+                fixed (byte* secret = _state.Secret)
+                {
+                    DigestLong(ref _state, accumulators, secret);
+                    current.Low64 = MergeAccumulators(accumulators, secret + SecretMergeAccsStartBytes, _state.TotalLength * Prime64_1);
+                    current.High64 = MergeAccumulators(accumulators, secret + SecretLengthBytes - AccumulatorCount * sizeof(ulong) - SecretMergeAccsStartBytes, ~(_state.TotalLength * Prime64_2));
+                }
+            }
+            else
+            {
+                fixed (byte* buffer = _state.Buffer)
+                {
+                    current = HashToHash128(new ReadOnlySpan<byte>(buffer, (int)_state.TotalLength), (long)_state.Seed);
+                }
+            }
+
+            return current;
+        }
+
+#if NET7_0_OR_GREATER
+        /// <summary>Gets the current computed hash value without modifying accumulated state.</summary>
+        /// <returns>The hash value for the data already provided.</returns>
+        [CLSCompliant(false)]
+        public UInt128 GetCurrentHashAsUInt128()
+        {
+            Hash128 current = GetCurrentHashAsHash128();
+            return new UInt128(current.High64, current.Low64);
+        }
+#endif
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static void WriteBigEndian128(in Hash128 hash, Span<byte> destination)
+        {
+            // TODO https://github.com/dotnet/runtime/issues/72107: Use BinaryPrimitives.WriteUInt128BigEndian() once it is supported
```

Comment:
We could, but since we're using a private Hash128, and we'd need to special-case it for different targets, and it's unlikely that will be measurably better an implementation, probably not very high value and we can delete the TODO.
```suggestion
```